### PR TITLE
Mnist experiment in nupic.vision

### PR DESCRIPTION
**Pull request by jaredweiss**
Moved the mnist experiment found in nupic.research/image_test to this repo (under nupicvision/examples). 
Merge after #18. 
Fixes: #13

### CODE REVIEW

**Comment by scottpurdy** [comment]
please separate into different PRs for ImageSensor updates and initial addition of MNIST example

**Comment by jaredweiss** [comment]
@scottpurdy Please review when you get a chance

File: nupicvision/examples/mnist/convertImages.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# ----------------------------------------------------------------------
+#  Copyright (C) 2006-2008 Numenta Inc. All rights reserved.
```
Comment (scottpurdy) [inline nupicvision/examples/mnist/convertImages.py]: wrong copyright header. Change to the open source version and update to 2006-2015

File: nupicvision/examples/mnist/mnist_extraction_source/extractMNIST.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
+// ----------------------------------------------------------------------
+//  Copyright (C) 2006-2008 Numenta Inc. All rights reserved.
```
Comment (scottpurdy) [inline nupicvision/examples/mnist/mnist_extraction_source/extractMNIST.cpp]: Similarly, need to switch to open source header and update the years

File: nupicvision/examples/mnist/run_mnist_experiment.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2014, Numenta, Inc.  Unless you have an agreement
```
Comment (scottpurdy) [inline nupicvision/examples/mnist/run_mnist_experiment.py]: year

**Comment by scottpurdy** [comment]
Should we just put it in `nupicvision/mnist`? If we have a bunch of different applications in the future then we can merge them into an examples directory but it seems like overkill at this point

File: nupicvision/examples/mnist/run_mnist_experiment.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+maxBoost                    = 1
+wrapAround                  = 1
+CPP SP seed                 = 1956
+
+"""
+
+
+
+DEFAULT_IMAGESENSOR_PARAMS ={
+  'width': 32,
```
Comment (scottpurdy) [inline nupicvision/examples/mnist/run_mnist_experiment.py]: not sure if you're doing style clean up in this but double quotes

**Comment by scottpurdy** [comment]
done reviewing

File: nupicvision/mnist/mnist_extraction_source/extractMNIST.cpp [hunk 5]
```diff
@@ -0,0 +1,223 @@
+// ----------------------------------------------------------------------
+// Numenta Platform for Intelligent Computing (NuPIC)
```
Comment (scottpurdy) [inline nupicvision/mnist/mnist_extraction_source/extractMNIST.cpp]: Take a look in nupic.core for the C++ format:
https://github.com/numenta/nupic.core/blob/master/src/nupic/math/DenseMatrix.hpp

**Comment by scottpurdy** [comment]
One minor comment and then I'm happy :smile:

**Pull request merged**